manim/mobject/text/tex_mobject.py

```diff
@@ -211,10 +211,13 @@ def get_tex_string(self):
         return self.tex_string
 
     def init_colors(self, propagate_colors=True):
-        if config.renderer == RendererType.OPENGL:
-            super().init_colors()
-        elif config.renderer == RendererType.CAIRO:
-            super().init_colors(propagate_colors=propagate_colors)
+        for submobject in self.submobjects:
```

Comment:
You might want to add a comment explaining that this is needed to preserve `\textcolor` colors, otherwise in the future someone might forget and revert it :)

Reply:
Thanks @JasonGrace2282, I've added something to both the docstring regarding the new functionality + comment for future reference. Feel free to suggest any edits to the wording of either if needed. :)

I also too the opportunity to slightly refactor the code from this PR to reduce a level of indentation and hopefully increase readability.